docs/site/Loopback-component-authorization.md

```diff
@@ -0,0 +1,547 @@
+    AUTHORIZATION_METADATA_ACCESSOR,
+    controllerClass.prototype,
+    methodName,
+  );
+}
+```
+
+{% include code-caption.html content="/src/authorization/providers/authorization-action.provider.ts" %}
+
+```ts
+import {Getter, inject, Provider} from '@loopback/context';
+
+import {AuthorizatonBindings} from '../keys';
+import {AuthorizationMetadata, AuthorizeFn} from '../types';
+
+import {intersection} from 'lodash';
+
+export class AuthorizeActionProvider implements Provider<AuthorizeFn> {
+  constructor(
+    @inject.getter(AuthorizatonBindings.METADATA)
+    private readonly getMetadata: Getter<AuthorizationMetadata>,
+  ) {}
+
+  value(): AuthorizeFn {
+    return response => this.action(response);
+  }
+
+  async action(userPermissions: string[]): Promise<boolean> {
+    const metadata: AuthorizationMetadata = await this.getMetadata();
+    if (!metadata) {
+      return false;
+    } else if (metadata.permissions.indexOf('*') === 0) {
+      // Return immediately with true, if allowed to all
+      // This is for publicly open routes only
+      return true;
+    }
+
+    // Add your own business logic to fetch or
+    // manipulate with user permissions here
+
+    const permissionsToCheck = metadata.permissions;
+    return intersection(userPermissions, permissionsToCheck).length > 0;
+  }
+}
+```
+
+Below is the user permissions manipulation logic. If there is no requirement of
+user level permissions in your application, you can skip the below.
+
+{% include code-caption.html content="/src/authorization/providers/user-permissions.provider.ts" %}
+
+```ts
+import {Provider} from '@loopback/context';
+
+import {PermissionKey} from '../permission-key';
+import {UserPermission, UserPermissionsFn} from '../types';
+
+export class UserPermissionsProvider implements Provider<UserPermissionsFn> {
+  constructor() {}
+
+  value(): UserPermissionsFn {
+    return (userPermissions, rolePermissions) =>
+      this.action(userPermissions, rolePermissions);
+  }
+
+  action(
+    userPermissions: UserPermission[],
+    rolePermissions: PermissionKey[],
+  ): PermissionKey[] {
+    let perms: PermissionKey[] = [];
+    // First add all permissions associated with role
+    perms = perms.concat(rolePermissions);
+    // Now update permissions based on user permissions
+    userPermissions.forEach((userPerm: UserPermission) => {
+      if (userPerm.allowed && perms.indexOf(userPerm.permission) < 0) {
+        // Add permission if it is not part of role but allowed to user
+        perms.push(userPerm.permission);
+      } else if (!userPerm.allowed && perms.indexOf(userPerm.permission) >= 0) {
+        // Remove permission if it is disallowed for user
+        perms.splice(perms.indexOf(userPerm.permission), 1);
+      }
+    });
+    return perms;
+  }
+}
+```
+
+Next, we need to expose these providers via Component to be bound to the
+context.
+
+{% include code-caption.html content="/src/authorization/component.ts" %}
```

Comment:
This code snippet still uses `Authoris*` instead of `Authoriz*`

Reply:
Fixed.